### PR TITLE
InstructorFeedbackEdit: Add a 'discard changes' button to question edit forms #1733

### LINKED ISSUE #1733: InstructorFeedbackEdit: Add a 'discard changes' button to question edit forms
_From [arnold.k...@gmail.com](https://code.google.com/u/105779801956403207546/) on March 21, 2014 11:50:44_

It is not currently clear how to "cancel" editing a question in the feedback edit page. Maybe can add a button to either reset the whole form and disable it or simply refresh the whole page?

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1736_

### PR DESCRIPTION
**Pull request by MrFool**
Fixes #1733

### CODE REVIEW

**Comment by MrFool** [comment]
I was in the middle of writing some tests for this but I found that there wasn't any test for delete new question's "delete" option for location.reload(), should I write tests for all question types testing cancel (click and confirm & click and cancel) for both new question and while editing?

**Comment by kanghj** [comment]
It seems unnecessary to have the test for all question types since this 'discard changes' button should be independent of question type. Perhaps just one for the new question form and one for an existing question?

**Comment by MrFool** [comment]
Lucky haven't write for all, would have bloated the file by a really huge amount of lines - ok noted :)

**Comment by MrFool** [comment]
Ok I've written the commit to do so with MCQ questions as the test cases - I'll run the tests tomorrow and if it's dev green I'll ping again and push it to origin, let you know again :)

**Comment by MrFool** [comment]
Will have to take a day more for this - apologies, underestimated the amount of changes needed to be made for the tests -> after adding the extra button, the HTML verification tests all broke

**Comment by MrFool** [comment]
Dev Green except #3315 which is fixed in #3315

**Comment by kanghj** [comment]
Looks like there are some new merge conflicts now, can you fix the conflicts first?

**Comment by wkurniawan07** [comment]
If you want to do a quick fix, just add this line:
`BackDoor.deleteCourse("CCAddUiTest.course1");`
before line 125 executes. Once it's a pass, you can remove it.

**Comment by MrFool** [comment]
Can't be sure that the tests are all green after merge, will test tomorrow and update when it's green

**Comment by MrFool** [comment]
Welps happen to be awake until the tests pass / failed - have updated it it's now green and have no more conflicts

File: src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java [hunk 1]
```diff
@@ -395,6 +395,20 @@ public WebElement getDeleteQuestionLink(int qnIndex){
...
+        return browser.driver.findElement(By.xpath("//a[@onclick='cancelEdit(" + qnIndex + ")']"));
+    }
+    
+    public boolean checkCancelEditQuestionButtonVisibility(int qnIndex) {
+        WebElement cancelEditButton = browser.driver.findElement(By.xpath("//a[@onclick='cancelEdit(" + qnIndex + ")']"));
+        
+        if (cancelEditButton.isDisplayed()) {
+            return true;
+        } else {
+            return false;
```
Comment (kanghj) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java]: return cancelEditButton.isDisplayed() ?

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 2]
```diff
@@ -259,6 +261,45 @@ private void testGetQuestionLink() {
...
+        // Add question 2 first
+        feedbackEditPage.selectNewQuestionType("Multiple-choice (single answer) question");
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.fillQuestionBox("mcq qn");
+        feedbackEditPage.fillMcqOption(0, "Choice 1");
+        feedbackEditPage.fillMcqOption(1, "Choice 2");
+        feedbackEditPage.clickAddQuestionButton();
+        
+        // Enable edit mode before testing canceling
+        assertEquals(true, feedbackEditPage.clickEditQuestionButton(1));
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: Use assertTrue instead?

**Comment by kanghj** [comment]
Some minor changes requested. Other than that, looks good.

**Comment by MrFool** [comment]
Ah right, those are good catches, let me fix those tonight :)

**Comment by MrFool** [comment]
Done, thanks for the nice catches ~

**Pull request merged**